python/ply_io.py

```diff
@@ -29,15 +29,22 @@ def read_ply(fp, newline=None):
             if 'element face' in line:
                 raise Exception('.ply appears to be a mesh')
             if 'end_header' in line: break
-    
+
+        for j in range(len(prop)):
+            if 'scalar_' in prop[j]:
+                prop[j] = prop[j].split('scalar_')[-1]
+        
         ply.seek(length)
 
         if fmt == 'binary':
             arr = np.fromfile(ply, dtype=','.join(dtype))
+            df = pd.DataFrame(data=arr)
+            df.columns = prop
+            df = df.dropna()  # drop rows with NAN values
         else:
-            arr = np.loadtxt(ply)
```

Comment:
Why are there NaN values?

Also this below is slightly more succinct?

        if fmt == 'binary':
            arr = np.fromfile(ply, dtype=','.join(dtype))
            df = pd.DataFrame(data=arr, columns=prop)
        else:
            df = pd.read_csv(ply, sep='\s+', names=prop, header=None)
           
        df = df.dropna()  # drop rows with NAN values

Reply:
The last point of an extracted cloud has NAN values in the last two columns (it does not have the last two properties), which caused filter_branch.py stopped running.

Yes, I agree. Moving df.dropna() out of the if statement is more succinct. Thanks.

Reply:
OK, how is the cloud being written in the first place before being read here as I think there must be an underlying issue. Dropping NaN here is a fix but could hide other errors e.g. what if it was the last 10,000 lines? I think I would put this outside the ply_io function.